pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -182,8 +182,8 @@ type CertificateSpec struct {
 	// a single `CertificateRequest` created by this Certificate, either when it
 	// was created, renewed, or Spec was changed. Revisions will be removed by
 	// oldest first if the number of revisions exceeds this number. If set,
-	// revisionHistoryLimit must be a value of `1` or greater. If unset (`nil`),
-	// revisions will not be garbage collected. Default value is `nil`.
+	// revisionHistoryLimit must be a value of `0` or greater. If set to `0`,
+	// revisions will not be garbage collected. Default value is `25`.
 	// +kubebuilder:validation:ExclusiveMaximum=false
 	// +optional
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"` // Validated by the validating webhook.
```

Comment:
Thanks for the reply. I agree that this might introduce unexpected behivour and like the idea to configure the default. We can do this either via the webhook or a cmd flag.